src/components/typography/text/text.js

```diff
@@ -33,14 +36,58 @@ const Headline = props => {
 
 Headline.displayName = 'TextHeadline';
 Headline.propTypes = {
-  elementType: PropTypes.oneOf(['h1', 'h2', 'h3']).isRequired,
+  as(props, propName, componentName, ...rest) {
+    // is required if the prop 'elementType' is not defined
+    if (!props.elementType) {
+      return PropTypes.oneOf(['h1', 'h2', 'h3']).isRequired(
+        props,
+        propName,
+        componentName,
+        ...rest
+      );
+    }
+    return PropTypes.oneOf(['h1', 'h2', 'h3'])(
```

Comment:
I'm a bit confused now. What's the purpose of changing the prop `elementType` to `as` if you have the same exact restrictions as before? 🤔

Reply:
...to settle on a convention most other component libraries also have in offeing a prop called `as` not `elementType`. Other components, before we started with #628 assuming there too, would then offer an `as` prop too if needed and not `elementType`.

Reply:
> To be honest, I think there are several implications that need to be discussed first instead of jumping into the implementation.

We have an issue where the prop name was suggested. So lets move the discussion there. Can you lay out the specific bigger implication you have for text components there?